src/stores/room-list/filters/NameFilterCondition.ts

```diff
@@ -19,16 +19,23 @@ import { FILTER_CHANGED, FilterKind, IFilterCondition } from "./IFilterCondition
 import { EventEmitter } from "events";
 import { normalize } from "matrix-js-sdk/src/utils";
 import { throttle } from "lodash";
+import { fuzzyMatch } from "../../../utils/strings";
+
+interface IOpts {
+    fuzzy?: boolean;
+}
 
 /**
  * A filter condition for the room list which reveals rooms of a particular
  * name, or associated name (like a room alias).
  */
 export class NameFilterCondition extends EventEmitter implements IFilterCondition {
     private _search = "";
+    private fuzzy = false;
 
-    constructor() {
+    constructor(options: IOpts = {}) {
```

Comment:
not sure how we feel about this more generally, but making this `private fuzzy = false` is slightly cleaner and works well with IDEs. The options approach does allow for expandability, however this is internal API and can be changed with relative ease.

Reply:
I'm not sure if I understand the change that you're suggesting.

As far as I can tell I have created `private fuzzy = false` and this options parameter is only here to let a developer choose the type of matching the `NameFilterCondition` should apply

Could you show me an example of what you had in mind instead?

Reply:
In this case, merging the field declaration with the constructor declaration: `constructor(private fuzzy = false) {`

It's a little bit cleaner for the API contract this part of the code is after.